eof/download.py

```diff
@@ -107,22 +91,56 @@ def download_eofs(orbit_dts=None, missions=None, sentinel_file=None, save_dir=".
             query.update(client.query_orbit_for_product(sentinel_file))
         else:
             for mission, dt in zip(missions, orbit_dts):
-                result = client.query_orbit(dt, dt + timedelta(days=1),
-                                            mission, product_type='AUX_POEORB')
+                found_result = False
+                products = client.query_orbit(dt - ScihubGnssClient.T0,
+                                              dt + ScihubGnssClient.T1,
+                                              mission,
+                                              product_type='AUX_POEORB')
+                result = (client._select_orbit(products, dt, dt + timedelta(minutes=1))
```

Comment:
This is part way, but not entirely solving, the point you made about consolidating the `lastval_cover` and other functions. But without calling the `_select_orbit`, it would download 2